core/src/test/java/de/danoeh/antennapod/core/feed/LocalFeedUpdaterTest.java

```diff
@@ -0,0 +1,258 @@
+            ShadowMediaMetadataRetriever.addMetadata(ds, MediaMetadataRetriever.METADATA_KEY_TITLE, file.getName());
+        }
+        shadowOf(MimeTypeMap.getSingleton()).addExtensionMimeTypMapping("mp3", "audio/mp3");
+    }
+
+    @SuppressWarnings("JavaReflectionMemberAccess")
+    @After
+    public void tearDown() throws Exception {
+        // Workaround for Robolectric issue in ShadowSQLiteConnection: IllegalStateException: Illegal connection pointer
+        Field field = PodDBAdapter.class.getDeclaredField("db");
+        field.setAccessible(true);
+        field.set(null, null);
+
+        for (Class<?> innerClass : PodDBAdapter.class.getDeclaredClasses()) {
+            if (innerClass.getSimpleName().equals("SingletonHolder")) {
+                Field dbHelperField = innerClass.getDeclaredField("dbHelper");
+                dbHelperField.setAccessible(true);
+                SQLiteOpenHelper dbHelper = (SQLiteOpenHelper) dbHelperField.get(null);
+                Field databaseField = SQLiteOpenHelper.class.getDeclaredField("mDatabase");
+                databaseField.setAccessible(true);
+                databaseField.set(dbHelper, null);
+            }
+        }
+    }
+
+    /**
+     * Test adding a new local feed.
+     */
+    @Test
+    public void testUpdateFeed_AddNewFeed() {
+        // verify empty database
+        List<Feed> feedListBefore = DBReader.getFeedList();
+        assertTrue(feedListBefore.isEmpty());
+
+        DocumentFile documentFile = DocumentFile.fromFile(localFeedDir2);
+        try (MockedStatic<DocumentFile> dfMock = Mockito.mockStatic(DocumentFile.class)) {
+            // mock external storage
+            dfMock.when(() -> DocumentFile.fromTreeUri(any(), any())).thenReturn(documentFile);
+
+            // call method to test
+            Feed feed = new Feed(FEED_URL, null);
+            LocalFeedUpdater.updateFeed(feed, context);
+        }
+
+        // verify new feed in database
+        List<Feed> feedListAfter = DBReader.getFeedList();
+        assertEquals(1, feedListAfter.size());
+        Feed feedAfter = feedListAfter.get(0);
+        assertEquals(FEED_URL, feedAfter.getDownload_url());
+        List<FeedItem> feedItems = DBReader.getFeedItemList(feedAfter);
+        assertEquals(2, feedItems.size());
+    }
+
+    /**
+     * Test adding further items to an existing local feed.
+     */
+    @Test
+    public void testUpdateFeed_AddMoreItems() {
+        // add local feed with 1 item (localFeedDir1)
+        DocumentFile documentFile1 = DocumentFile.fromFile(localFeedDir1);
+        try (MockedStatic<DocumentFile> dfMock = Mockito.mockStatic(DocumentFile.class)) {
+            // mock external storage
+            dfMock.when(() -> DocumentFile.fromTreeUri(any(), any())).thenReturn(documentFile1);
+            Feed feed = new Feed(FEED_URL, null);
+            LocalFeedUpdater.updateFeed(feed, context);
+        }
```

Comment:
Adding items seems to be pretty repetitive code. How about creating a function for that?

Reply:
Do you mean to extract a helper method with just the following two lines?

```
Feed feed = new Feed(FEED_URL, null);
LocalFeedUpdater.updateFeed(feed, context);
```

Reply:
I meant the whole thing about mocking external storage and updating the feed. That looks pretty similar for all tests (I have only skimmed the code, though. Maybe I have overlooked something).

```
	DocumentFile documentFile = DocumentFile.fromFile(localFeedDir2);
        try (MockedStatic<DocumentFile> dfMock = Mockito.mockStatic(DocumentFile.class)) {
            // mock external storage
            dfMock.when(() -> DocumentFile.fromTreeUri(any(), any())).thenReturn(documentFile);

            // call method to test
            Feed feed = new Feed(FEED_URL, null);
            LocalFeedUpdater.updateFeed(feed, context);
        }
```

Reply:
Ok, I see what you mean. I extracted repetitive code into helper methods, for mocking external storage as mentioned above, and also the checks for the number of feeds and the number of items in the feed. The test methods look very simple now.